priv/script.sh

```diff
@@ -2,441 +2,490 @@
 
 set -eu
 
+elixir_version=1.17.3-otp-27
+erlang_version=27.1.2
+phoenix_version=1.7.14
+postgres_version=15.1
+
+OS="$(uname -s)"
+case "$OS" in
+Linux*) os_type=Linux ;;
+Darwin*) os_type=macOS ;;
+*)
+  printf >&2 "Unsupported OS: %s\n" "$OS"
+  exit 1
+  ;;
+esac
+
+has() {
+  command -V "${1:?}" >/dev/null 2>/dev/null
+}
+
+mise_has() {
+  mise which "${1:?}" >/dev/null 2>/dev/null
+}
+
+mise_has_version() {
+  has mise && mise_has "${1:?}" && case "$(mise which --version "${1:?}")" in
+  "${2:?}"*) : ;;
+  *) return 1 ;;
+  esac
+}
+
 # Make sure important variables exist if not already defined
 #
 # $USER is defined by login(1) which is not always executed (e.g. containers)
 # POSIX: https://pubs.opengroup.org/onlinepubs/009695299/utilities/id.html
 USER=${USER:-$(id -u -n)}
+
 # $HOME is defined at the time of login, but it could be unset. If it is unset,
 # a tilde by itself (~) will not be expanded to the current user's home directory.
 # POSIX: https://pubs.opengroup.org/onlinepubs/009696899/basedefs/xbd_chap08.html#tag_08_03
-HOME="${HOME:-$(getent passwd $USER 2>/dev/null | cut -d: -f6)}"
-# macOS does not have getent, but this works even if $HOME is unset
-HOME="${HOME:-$(eval echo ~$USER)}"
-
-bold='\033[1m'
-normal='\033[0m'
-red='\033[0;31m'
-blue='\033[0;34m'
-bblue='\033[1;34m'
-white='\033[0;37m'
-green='\033[0;32m'
-cyan='\033[0;36m'
-
-elixir_version=1.17.2-otp-27
-erlang_version=27.0.1
-phoenix_version=1.7.14
-postgres_version=15.1
+if [ -z "$HOME" ]; then
+  set +e
+  if [ "$os_type" = Linux ]; then
+    HOME="$(getent passwd "$USER" 2>/dev/null | cut -d: -f6)"
+  elif [ "$os_type" = macOS ]; then
+    HOME="$(dscl . -read /Users/"$USER" NFSHomeDirectory)"
```

Comment:
This is how you read the home directory on macOS. It means that the fallback on line 53 is less necessary in general.